src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,243 @@
+using AutoRest.CSharp.Common.Output.Models.Types;
+using AutoRest.CSharp.Common.Output.Models.Types.HelperTypeProviders;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+
+namespace AutoRest.CSharp.Common.Output.Builders
+{
+    internal class MultipartSerializationMethodsBuilder
+    {
+        public static IEnumerable<Method> BuildMultipartSerializationMethods(MultipartObjectSerialization multipart)
+        {
+            /* private BinaryData SerializeMultipart(ModelReaderWriterOptions options) */
+            yield return new Method(
+                new MethodSignature(
+                    "SerializeMultipart",
+                    null,
+                    null,
+                    MethodSignatureModifiers.Private,
+                    typeof(BinaryData),
+                    null,
+                    new Parameter[] { KnownParameters.Serializations.Options }),
+                BuildMultipartSerializationMethodBody().ToArray());
+
+            /* internal virtual MultipartFormDataRequestContent ToMultipartRequestContent() */
+            yield return new Method(
+                new MethodSignature(
+                    Configuration.ApiTypes.ToMultipartRequestContentName,
+                    null,
+                    null,
+                    MethodSignatureModifiers.Internal | MethodSignatureModifiers.Virtual,
+                    MultipartFormDataRequestContentProvider.Instance.Type,
+                    null,
+                    Array.Empty<Parameter>()),
+                BuildToMultipartRequestContentMethodBody(multipart).ToArray());
+        }
+        public static SwitchCase BuildMultipartWriteSwitchCase(MultipartObjectSerialization multipart, ModelReaderWriterOptionsExpression options)
+        {
+            return new SwitchCase(
+                Serializations.MultipartFormat,
+                Return(new InvokeInstanceMethodExpression(null, "SerializeMultipart", new[] { options }, null, false)));
+        }
+
+        public static IEnumerable<MethodBodyStatement> BuildMultipartSerializationMethodBody()
+        {
+            var serializeCallExpression = new InvokeInstanceMethodExpression(null, Configuration.ApiTypes.ToMultipartRequestContentName, Array.Empty<ValueExpression>(), null, false);
+            return new MethodBodyStatement[]
+                        {
+                            UsingDeclare("content", MultipartFormDataRequestContentProvider.Instance.Type, serializeCallExpression, out var content),
+                            /*using MemoryStream stream = new MemoryStream();*/
+                            UsingDeclare("stream", typeof(MemoryStream), New.Instance(typeof(MemoryStream), Array.Empty<ValueExpression>()), out var stream),
```

Comment:
there is an overload that we could have strong-typed expressions:
```
public static MethodBodyStatement UsingDeclare(string name, StreamExpression value, out StreamExpression variable)
```
therefore 
```suggestion
                            UsingDeclare("stream", new StreamExpression(New.Instance(typeof(MemoryStream), Array.Empty<ValueExpression>())), out var stream), // <- now stream is StreamExpression
```